packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3409,6 +3382,36 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     return Actions.invoke(context, intent);
   }
 
+  // Tracks the location a [_ScribblePlaceholder] should be rendered in the
+  // text.
+  //
+  // A value of -1 indicates there should be no placeholder, otherwise the
+  // value should be between 0 and the length of the text, inclusive.
+  int _placeholderLocation = -1;
+
+  void _onPlaceholderLocationChanged(int location) {
+    setState(() {
+      _placeholderLocation = location;
+    });
+  }
+
+  void _onScribbleFocus(Offset offset) {
+    widget.focusNode.requestFocus();
+    renderEditable.selectPositionAt(from: offset, cause: SelectionChangedCause.scribble);
+    _openInputConnection();
+    _updateSelectionRects(force: true);
+  }
+
+  void _removeTextPlaceholder() {
+    if (!widget.scribbleEnabled) {
```

Comment:
shouldn't the same check be performed in other methods as well?

Reply:
In _onPlaceholderLocationChanged and _onScribbleFocus, the `_ScribbleFocusable` widget prevents them from being called when scribble is disabled.  Should I add the check (or an assert) just as a double check you think?